proposals/solid-notifications.md

```diff
@@ -0,0 +1,270 @@
+
+The client sends a JSON-LD payload to this endpoint via POST. The only required fields in this interaction are type and topic. The type field MUST contain the type of subscription being requested. The topic field MUST contain the URL of the resource a client wishes to subscribe to changes.
+
+All other fields are defined by the particular subscription type. Some common features are described in the Features section of this document.
+
+An example POST request using a DPoP bound access token is below:
+
+```http
+POST /subscription
+Authorization: DPoP <token>
+DPoP: <proof>
+Content-Type: application/ld+json
+
+{
+    "@context": ["https://www.w3.org/ns/solid/notification/v1"],
+    "type": "WebSocketSubscription2021",
+    "topic": "https://pod.example/resource",
+    "state": "opaque-state",
+    "expiration": "2021-09-21T12:37:15Z",
+    "rate": "PT10s"
+}
+```
+
+A successful response will contain a URL that can be used directly with a JavaScript client
+
+```http
+HTTP/2
+Content-Type: application/ld+json
+
+{
+    "@context": "https://www.w3.org/ns/solid/notification/v1",
+    "type": "WebSocketSubscription2021",
+    "endpoint": "wss://websocket.example/?auth=Ys3KiUq"
+}
+```
+
+In JavaScript, a client can use the data in the response to establish a connection to the WebSocket endpoint.
+
+```javascript
+const ws = new WebSocket(endpoint, type)
+```
+
+A client will define how to handle events, especially the onclose and onmessage events.
+
+```javascript
+ws.onclose(evt => reconnect(evt));
+ws.onmessage(evt => console.log("Message received: ", evt))
+```
+
+## Authentication and Authorization
+
+The Subscription API requires authorization. This document does not define the specific technology used to authorize requests; rather, it defers to the Solid Protocol sections on Authentication and Authorization. It is out of scope for this document to describe how a client fetches an authorization token. Solid-OIDC is one example of an authentication mechanism that could be used with Solid Notifications.
+
+Resource-based Access controls MUST be enforced for every subscription. A client MUST be permitted to read the resource to which it subscribes.
+
+## Notification Data Model
+
+The content of a Solid Notification is defined in terms of a W3C Activity Streams 2.0 document. As such, JSON-LD 1.0 is a required baseline serialization format, though other formats are not prohibited. JSON-LD version 1.1 is recommended when using JSON-LD serializations.
+
+```javascript
+{
+   "@context":[
+      "https://www.w3.org/ns/activitystreams",
+      "https://www.w3.org/ns/solid/notification/v1"
+   ],
+   "id":"urn:uuid:<uuid>",
+   "type":[
+     "Update"
+   ],
+   "actor":[
+      "<WebID>"
+   ],
+   "object":{
```

Comment:
Suggest a level 2 protocol in which the triples which changed in an RDF resource are included in the notification.

Reply:
Noted. The contents of a notification will be discussed and deserves its own thread of conversation. Does this concern block merging this PR in its present form?